Create initial internal `pip_api` package (#2286)
jazzband/pip-tools: changelog.d/2285.contrib.md

> Utilities for interacting with `pip` have started to move into an internal `_pip_api` subpackage -- by {user}`sirosen`.

Reviewer: I suppose its
```suggestion
Utilities for interacting with `pip` have started to move into the :py:mod:`piptools._internal._pip_api` subpackage -- by {user}`sirosen`.
```
now?
